client/src/partials/general_ledger/general_ledger.js

```diff
@@ -0,0 +1,137 @@
+angular.module('bhima.controllers')
+  .controller('GeneralLedgerController', GeneralLedgerController);
+
+GeneralLedgerController.$inject = [
+  'GeneralLedgerService', 'GridSortingService', 'GridGroupingService',
+  'GridFilteringService', 'GridColumnService', 'JournalConfigService',
+  'SessionService', 'NotifyService', 'TransactionService'
+];
+
+/**
+ * @module GeneralLedgerController
+ *
+ * @description
+ * This controller is responsible for initialising the core client side general ledger,
+ * binding the UI Grid component with services that facilitate all
+ * operations required by an accountant without editing.
+ * - Displaying transactions in an easy to find and review format
+ *   - Search for transactions
+ *   - Filter transactions
+ *   - Group by transactions to show aggregates
+ *   - Sort transactions
+ *   - Show or hide columns
+ *
+ */
+function GeneralLedgerController(GeneralLedger, Sorting, Grouping, Filtering, Columns, Config, Session, Notify, Transactions) {
+  var vm = this;
+
+  // General Ledger utilities
+  var sorting, grouping, filtering, columnConfig, transactions;
+
+  /** @const the cache alias for this controller */
+  var cacheKey = 'GeneralLedger';
+
+  vm.enterprise = Session.enterprise;
+
+  // gridOptions is bound to the UI Grid and used to configure many of the
+  // options, it is also used by the grid to expose the API
+  vm.gridOptions = {
```

Comment:
Since this is the General Ledger and no editing will be made on the data, you should try using the options `fastWatch` and or `flatEntityAccess` to increase performance.  [Src](http://ui-grid.info/docs/#/tutorial/404_large_data_sets_and_performance)